enhancements/kube-apiserver/stability-point-to-point-network-check.md

```diff
@@ -0,0 +1,418 @@
+	// Status of the condition
+	Status metav1.ConditionStatus `json:"status"`
+	// machine readable reason for the condition's last transition.
+	Reason string `json:"reason,omitempty"`
+	// human readable message indicating details about last transition.
+	Message string `json:"message,omitempty"`
+	// Last time the condition transit from one status to another.
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+}
+
+type LogEntryReason string
+
+type PodNetworkConnectivityCheckConditionType string
+
+const (
+	Reachable    PodNetworkConnectivityCheckConditionType = "Reachable"
+	DNSDone      LogEntryReason                           = "DNSDone"
+	DNSError     LogEntryReason                           = "DNSError"
+	ConnectDone  LogEntryReason                           = "ConnectDone"
+	ConnectError LogEntryReason                           = "ConnectError"
+)
+
+```
+
+
+#### Point-to-point Network Check Tool
+
+Runs in a container in the interested pod. Executes check and updates `PodNetworkConnectivityCheck` resources for the pod.
+
+Usage: 
+
+```
+p2pnc check tcp-endpoint
+
+``` 
+
+
+### Risks and Mitigations
+
+What are the risks of this proposal and how do we mitigate. Think broadly. For
+example, consider both security and how this will impact the larger OKD
+ecosystem.
+
+How will security be reviewed and by whom? How will UX be reviewed and by whom?
+
+Consider including folks that also work outside your immediate sub-project.
+
+## Design Details
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, or as something else. Initial proposal
+should keep this high-level with a focus on what signals will be looked at to
+determine graduation.
+
+Consider the following in developing the graduation criteria for this
+enhancement:
+- Maturity levels - `Dev Preview`, `Tech Preview`, `GA`
+- Deprecation
+
+Clearly define what graduation means.
+
+#### Examples
+
+##### Example PodNetworkConnectivityCheck instance
+
+```yaml
+kind: PodNetworkConnectivityCheck
+version: network.openshift.io/v1alpha1
+metadata:
+    name: kas-10-0-137-239-r7-to-etcd-10-0-148-99
+    namespace: openshift-kube-apiserver
+spec:
+    sourcePodName: kube-apiserver-ip-10-0-135-158.us-west-1.compute.internal
+    sourcePodSelector:
```

Comment:
missing from api definition.

I would not suggest this field.  Update to these pods rare impact connectivity

Reply:
Removed.